modules/tensor_mechanics/src/userobjects/CrackFrontDefinition.C

```diff
@@ -129,9 +129,11 @@ CrackFrontDefinition::CrackFrontDefinition(const InputParameters & parameters)
     _q_function_type(getParam<MooseEnum>("q_function_type")),
     _crack_front_points_provider(nullptr)
 {
+  auto boundary = isParamValid("boundary") ? getParam<std::vector<BoundaryName>>("boundary")
+                                           : std::vector<BoundaryName>{};
```

Comment:
default in boundary restrictable could work. It might bring a lot of code to adapt though

Reply:
I tried but default won't work. It failed the cases below: Mainly because "crack_front_points" and "crack_front_points_provider" uses "boundary" differently
`test:crack_tip_enrichment.crack_tip_enrichment_edge_2d ....................................... FAILED (ERRMSG)
test:solid_mechanics_basic.edge_crack_3d ..................................................... FAILED (ERRMSG)
test:solid_mechanics_basic.penny_crack ....................................................... FAILED (ERRMSG)


test:nucleation_uo.nucleateCrack ............................................................. FAILED (ERRMSG)
test:nucleation_uo.filterNucleation .......................................................... FAILED (ERRMSG)
test:solid_mechanics_basic.edge_crack_3d_fatigue ............................................. FAILED (ERRMSG)
test:solid_mechanics_basic.edge_crack_3d_mhs ................................................. FAILED (ERRMSG)
test:solid_mechanics_basic.edge_crack_2d_propagation_mhs ..................................... FAILED (ERRMSG)
test:solid_mechanics_basic.elliptical_crack .................................................. FAILED (ERRMSG)
test:solid_mechanics_basic.penny_crack_cfp ................................................... FAILED (ERRMSG)`

Reply:
ok that was not unexpected. people have just been coding assuming no default downstream, and their code errors with 
`boundary = ""` already